src/sentry/eventstore/base.py

```diff
@@ -0,0 +1,143 @@
+from __future__ import absolute_import
+
+from enum import Enum
+
+from sentry import nodestore
+from sentry.utils.services import Service
+
+
+class Columns(Enum):
+    event_id = 'event_id'
+    group_id = 'group_id'
+    project_id = 'project_id'
+    timestamp = 'timestamp'
+    culprit = 'culprit'
+    location = 'location'
+    message = 'message'
+    platform = 'platform'
+    title = 'title'
+    type = 'type'
+    tags_key = 'tags.key'
+    tags_value = 'tags.value'
+    email = 'email'
+    ip_address = 'ip_address'
+    user_id = 'user_id'
+    username = 'username'
+
+
+class EventStorage(Service):
+    __all__ = (
+        'minimal_columns',
+        'full_columns',
+        'get_event_by_id',
+        'get_events',
+        'get_prev_event_id',
+        'get_next_event_id',
+        'bind_nodes',
+    )
+
+    # The minimal list of columns we need to get from snuba to bootstrap an
+    # event. If the client is planning on loading the entire event body from
+    # nodestore anyway, we may as well only fetch the minimum from snuba to
+    # avoid duplicated work.
+    minimal_columns = [
+        Columns.event_id,
+        Columns.group_id,
+        Columns.project_id,
+        Columns.timestamp,
+    ]
+
+    # A list of all useful columns we can get from snuba.
+    full_columns = minimal_columns + [
+        Columns.culprit,
+        Columns.location,
+        Columns.message,
+        Columns.platform,
+        Columns.title,
+        Columns.type,
+
+        # Required to provide snuba-only tags
+        Columns.tags_key,
+        Columns.tags_value,
+
+        # Required to provide snuba-only 'user' interface
+        Columns.email,
+        Columns.ip_address,
+        Columns.user_id,
+        Columns.username,
+    ]
+
+    def get_event_by_id(self, project_id, event_id, cols):
+        """
+        Gets a single event given a project_id and event_id.
+
+        Keyword arguments:
+        project_id (int): Project ID - default None
+        event_id (str): Event ID - default None
+        cols: (List[str]) - List of columns to fetch - default minimal_columns
```

Comment:
On second thoughts, I think one option would be to always fetch the minimal_columns (AFAIK we just never really want to fetch less than that) and have this be a list of additional columns for the query.

Reply:
I prefer this because then the user of this function can never forget to fetch the fields required for the composite primary key and get in trouble because of that later.